internal/worker/server.go

```diff
@@ -299,58 +327,87 @@ func (h *apiHandlers) RequestJob(ctx echo.Context) error {
 
 	jobId, token, jobType, jobArgs, dynamicJobArgs, err := h.server.RequestJob(ctx.Request().Context(), body.Arch, body.Types)
 	if err != nil {
-		return err
+		return api.HTTPErrorWithInternal(api.ErrorRequestingJob, err)
+	}
+
+	// TODO this juggling can be removed with deepmap>=1.8 and `x-oapi-codegen-extra-tags`
+	// and marking them as required but omitempty
+	var respArgs *json.RawMessage
+	if len(jobArgs) != 0 {
+		respArgs = &jobArgs
+	}
+	var respDynArgs *[]json.RawMessage
+	if len(dynamicJobArgs) != 0 {
+		respDynArgs = &dynamicJobArgs
 	}
 
-	return ctx.JSON(http.StatusCreated, requestJobResponse{
-		Id:               jobId,
+	response := api.RequestJobResponse{
+		ObjectReference: api.ObjectReference{
+			Href: fmt.Sprintf("%s/jobs", api.BasePath),
+			Id:   jobId.String(),
+			Kind: "RequestJob",
+		},
 		Location:         fmt.Sprintf("%s/jobs/%v", api.BasePath, token),
 		ArtifactLocation: fmt.Sprintf("%s/jobs/%v/artifacts/", api.BasePath, token),
 		Type:             jobType,
-		Args:             jobArgs,
-		DynamicArgs:      dynamicJobArgs,
-	})
+		Args:             respArgs,
+		DynamicArgs:      respDynArgs,
+	}
+	return ctx.JSON(http.StatusCreated, response)
 }
 
 func (h *apiHandlers) GetJob(ctx echo.Context, tokenstr string) error {
 	token, err := uuid.Parse(tokenstr)
 	if err != nil {
-		return echo.NewHTTPError(http.StatusBadRequest, "cannot parse job token")
+		return api.HTTPError(api.ErrorMalformedJobToken)
 	}
 
 	jobId, err := h.server.jobs.IdFromToken(token)
 	if err != nil {
 		switch err {
 		case jobqueue.ErrNotExist:
-			return ErrInvalidToken
+			return api.HTTPError(api.ErrorJobNotFound)
 		default:
-			return err
+			return api.HTTPError(api.ErrorResolvingJobId)
 		}
 	}
 
+	// TODO this should return 404? why would jobId ever be nil here
 	if jobId == uuid.Nil {
-		return ctx.JSON(http.StatusOK, getJobResponse{})
+		return ctx.JSON(http.StatusOK, api.GetJobResponse{
+			ObjectReference: api.ObjectReference{
+				Href: fmt.Sprintf("%s/jobs/%v", api.BasePath, token),
+				Id:   token.String(),
+				Kind: "JobStatus",
+			},
+			Canceled: false,
+		})
```

Comment:
Was mostly confused about this part.

Reply:
I agree with keeping the API. Maybe open an issue instead and resolve this conversation (unless someone has an explanation that is, I don't :) )

Reply:
https://github.com/osbuild/osbuild-composer/issues/1824